Index: app/src/main/java/app/com/lily/tanvir/sunshine_verson_2/DetailActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/app/com/lily/tanvir/sunshine_verson_2/DetailActivity.java	(date 1481634095000)
+++ app/src/main/java/app/com/lily/tanvir/sunshine_verson_2/DetailActivity.java	(revision )
@@ -84,9 +84,11 @@
             return rootView;
         }
 
+
+
         private Intent createShareForecastIntent(){
             Intent shareIntent=new Intent(Intent.ACTION_SEND);
-            shareIntent.setFlags(Intent.FLAG_GRANT_WRITE_URI_PERMISSION);
+            shareIntent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_WHEN_TASK_RESET);
             shareIntent.putExtra(Intent.EXTRA_TEXT,forecastStr+HASHTAG);
             shareIntent.setType("text/plain");
             return shareIntent;
@@ -95,6 +97,17 @@
         @Override
         public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
             super.onCreateOptionsMenu(menu, inflater);
+            inflater.inflate(R.menu.detailfragment,menu);
+
+            MenuItem menuItem=menu.findItem(R.id.action_share);
+
+            shareActionProvider=(ShareActionProvider)MenuItemCompat.getActionProvider(menuItem);
+
+            if(shareActionProvider!=null){
+                shareActionProvider.setShareIntent(createShareForecastIntent());
+            }else {
+                Log.d(LOG_TAG,"Share action provider is null?");
+            }
 
         }
     }
